src/compiler/core.ts

```diff
@@ -2649,11 +2645,15 @@ namespace ts {
         readonly directories: string[];
     }
 
-    export function createCachedPartialSystem(host: HostForCaching): CachedPartialSystem {
+    export function createCachedPartialSystem(host: PartialSystem): CachedPartialSystem {
         const cachedReadDirectoryResult = createMap<MutableFileSystemEntries>();
         const getCurrentDirectory = memoize(() => host.getCurrentDirectory());
         const getCanonicalFileName = createGetCanonicalFileName(host.useCaseSensitiveFileNames);
         return {
+            useCaseSensitiveFileNames: host.useCaseSensitiveFileNames,

```

Comment:
We might want to use `() => host.useCaseSensitiveFileNames()` to avoid unbound `this` errors.